llvm/lib/ExecutionEngine/JITLink/aarch32.cpp

```diff
@@ -326,26 +359,16 @@ Expected<int64_t> readAddendData(LinkGraph &G, Block &B, const Edge &E) {
 Expected<int64_t> readAddendArm(LinkGraph &G, Block &B, const Edge &E) {
   ArmRelocation R(B.getContent().data() + E.getOffset());
   Edge::Kind Kind = E.getKind();
+  if (!checkOpcode(R, Kind))
+      return makeUnexpectedOpcodeError(G, R, Kind);
```

Comment:
We will get the dynamic opcode check here and get rid of the "templated" checks below. I think it significantly reduces noise and allows for much better grouping of case labels. What do you think?